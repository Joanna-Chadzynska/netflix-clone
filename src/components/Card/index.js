import React, { createContext, useContext, useState } from 'react';
import {
	Container,
	Entities,
	Feature,
	Group,
	Image,
	Item,
	Meta,
	SubTitle,
	Text,
	Title,
} from './styles/card';

export const FeatureContext = createContext();

const Card = ({ children, ...restProps }) => {
	const [showFeature, setShowFeature] = useState(false);
	const [itemFeature, setItemFeature] = useState({});

	return (
		<FeatureContext.Provider
			value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}>
			<Container {...restProps}>{children}</Container>
		</FeatureContext.Provider>
	);
};

Card.Group = function CardGroup({ children, ...restProps }) {
	return <Group {...restProps}>{children}</Group>;
};

Card.Title = function CardTitle({ children, ...restProps }) {
	return <Title {...restProps}>{children}</Title>;
};

Card.SubTitle = function CardSubTitle({ children, ...restProps }) {
	return <SubTitle {...restProps}>{children}</SubTitle>;
};

Card.Text = function CardText({ children, ...restProps }) {
	return <Text {...restProps}>{children}</Text>;
};

Card.Meta = function CardMeta({ children, ...restProps }) {
	return <Meta {...restProps}>{children}</Meta>;
};

Card.Item = function CardItem({ item, children, ...restProps }) {
	const { setItemFeature, setShowFeature } = useContext(FeatureContext);
	const handleFeature = () => {
		setItemFeature(item);
		setShowFeature(true);
	};
	return (
		<Item {...restProps} onClick={handleFeature}>
			{children}
		</Item>
	);
};

Card.Entities = function CardEntities({ children, ...restProps }) {
	return <Entities {...restProps}>{children}</Entities>;
};

Card.Image = function CardImage({ ...restProps }) {
	return <Image {...restProps} />;
};

Card.Feature = function CardFeature({ children, ...restProps }) {
	const { showFeature, itemFeature, setShowFeature } = useContext(
		FeatureContext
	);
	return <Feature {...restProps}>{children}</Feature>;
};

export default Card;
