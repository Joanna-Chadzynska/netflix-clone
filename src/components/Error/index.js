import React from 'react';
import {
	Background,
	ButtonLink,
	Code,
	Container,
	Content,
	ImageSource,
	Inner,
	RequestInfo,
	SubTitle,
	Title,
} from './styles/error';

const Error = ({ children, ...restProps }) => {
	return (
		<Background>
			<Container {...restProps}>{children}</Container>
		</Background>
	);
};

Error.Inner = function ErrorInner({ children, ...restProps }) {
	return <Inner {...restProps}>{children}</Inner>;
};

Error.Content = function ErrorContent({ children, ...restProps }) {
	return <Content {...restProps}>{children}</Content>;
};

Error.Title = function ErrorTitle({ children, ...restProps }) {
	return <Title {...restProps}>{children}</Title>;
};

Error.SubTitle = function ErrorSubTitle({ children, ...restProps }) {
	return <SubTitle {...restProps}>{children}</SubTitle>;
};

Error.ButtonLink = function ErrorButtonLink({ children, ...restProps }) {
	return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Error.Code = function ErrorCode({ children, ...restProps }) {
	return <Code {...restProps}>{children}</Code>;
};

Error.ImageSource = function ErrorImageSource({ children, ...restProps }) {
	return <ImageSource {...restProps}>{children}</ImageSource>;
};

Error.RequestInfo = function ErrorRequestInfo({ children, ...restProps }) {
	return <RequestInfo {...restProps}>{children}</RequestInfo>;
};

export default Error;
