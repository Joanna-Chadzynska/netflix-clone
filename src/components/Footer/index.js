import React from 'react';
import {
	Break,
	Column,
	Container,
	LanguageLabel,
	LanguageOpt,
	LanguagePicker,
	LanguageSelectContainer,
	Link,
	Row,
	SelectOpt,
	Text,
	Title,
} from './styles/footer';

const Footer = ({ children, ...restProps }) => {
	return <Container {...restProps}>{children}</Container>;
};

Footer.Row = function FooterRow({ children, ...restProps }) {
	return <Row {...restProps}>{children}</Row>;
};

Footer.Column = function FooterColumn({ children, ...restProps }) {
	return <Column {...restProps}>{children}</Column>;
};

Footer.Link = function FooterLink({ children, ...restProps }) {
	return <Link {...restProps}>{children}</Link>;
};

Footer.Title = function FooterTitle({ children, ...restProps }) {
	return <Title {...restProps}>{children}</Title>;
};

Footer.Text = function FooterText({ children, ...restProps }) {
	return <Text {...restProps}>{children}</Text>;
};

Footer.Break = function FooterBreak({ ...restProps }) {
	return <Break {...restProps} />;
};

Footer.LanguagePicker = function FooterLanguagePicker({ ...restProps }) {
	return <LanguagePicker {...restProps} />;
};

Footer.LanguageLabel = function FooterLanguageLabel({ ...restProps }) {
	return <LanguageLabel {...restProps} />;
};

Footer.LanguageSelectContainer = function FooterLanguageSelectContainer({
	...restProps
}) {
	return <LanguageSelectContainer {...restProps} />;
};

Footer.Select = function FooterSelect({ ...restProps }) {
	return <SelectOpt {...restProps} />;
};

Footer.LanguageOpt = function FooterLanguageOpt({ ...restProps }) {
	return <LanguageOpt {...restProps} />;
};

export default Footer;
