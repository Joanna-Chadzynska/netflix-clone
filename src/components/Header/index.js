import React, { useEffect, useState } from 'react';
import { Link as ReachRouterLink } from 'react-router-dom';
import { useClickOutside } from './../../hooks';
import {
	Background,
	ButtonLink,
	Container,
	Dropdown,
	Feature,
	FeatureCallOut,
	Group,
	Link,
	Logo,
	Picture,
	PlayButton,
	Profile,
	Search,
	SearchIcon,
	SearchInput,
	Text,
} from './styles/header';

const Header = ({ bg = true, children, ...restProps }) => {
	return bg ? <Background {...restProps}>{children}</Background> : children;
};

Header.Feature = function HeaderFeature({ children, ...restProps }) {
	return <Feature {...restProps}>{children}</Feature>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
	return <Container {...restProps}>{children}</Container>;
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
	return <Group {...restProps}>{children}</Group>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
	return (
		<ReachRouterLink to={to}>
			<Logo {...restProps} />
		</ReachRouterLink>
	);
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
	return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.PlayButton = function HeaderPlayButton({ children, ...restProps }) {
	return <PlayButton {...restProps}>{children}</PlayButton>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
	return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
	return <Link {...restProps}>{children}</Link>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
	children,
	...restProps
}) {
	return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
	return <Profile {...restProps}>{children}</Profile>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
	return <Picture src={`/images/users/${src}.png`} {...restProps} />;
};

Header.Dropdown = function HeaderDropdown({ src, ...restProps }) {
	return <Dropdown src={`/images/users/${src}.png`} {...restProps} />;
};

Header.Search = function HeaderSearch({
	searchTerm,
	setSearchTerm,
	...restProps
}) {
	const { ref, clickOutside } = useClickOutside();
	const [searchActive, setSearchActive] = useState(false);
	const handleClickActive = () => setSearchActive(!searchActive);

	useEffect(() => {
		if (clickOutside) {
			setSearchActive(false);
		}
	}, [clickOutside]);

	return (
		<Search {...restProps} ref={ref}>
			<SearchIcon onClick={handleClickActive}>
				<img src='/images/icons/search.png' alt='Search' />
			</SearchIcon>
			<SearchInput
				type='search'
				value={searchTerm}
				onChange={(e) => setSearchTerm(e.target.value)}
				placeholder='Search films and series'
				active={searchActive}
			/>
		</Search>
	);
};

export default Header;
