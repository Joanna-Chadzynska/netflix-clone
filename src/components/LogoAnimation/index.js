import React from 'react';
import {
	Container,
	Curve,
	Fill,
	LetterE,
	LetterF,
	LetterI,
	LetterL,
	LetterN,
	LetterT,
	LetterX,
	Shape,
	Word,
} from './styles/logoAnimation';

const LogoAnimation = ({ children, ...restProps }) => {
	return (
		<Container {...restProps}>
			<Word>
				{children}
				<Curve />
			</Word>
		</Container>
	);
};

LogoAnimation.LetterN = function LogoAnimationLetterN({
	children,

	...restProps
}) {
	return <LetterN {...restProps}>{children}</LetterN>;
};

LogoAnimation.LetterE = function LogoAnimationLetterN({
	children,

	...restProps
}) {
	return <LetterE {...restProps}>{children}</LetterE>;
};

LogoAnimation.LetterT = function LogoAnimationLetterN({
	children,

	...restProps
}) {
	return <LetterT {...restProps}>{children}</LetterT>;
};

LogoAnimation.LetterF = function LogoAnimationLetterN({
	children,

	...restProps
}) {
	return <LetterF {...restProps}>{children}</LetterF>;
};

LogoAnimation.LetterL = function LogoAnimationLetterN({
	children,

	...restProps
}) {
	return <LetterL {...restProps}>{children}</LetterL>;
};

LogoAnimation.LetterI = function LogoAnimationLetterN({
	children,

	...restProps
}) {
	return <LetterI {...restProps}>{children}</LetterI>;
};

LogoAnimation.LetterX = function LogoAnimationLetterN({
	children,

	...restProps
}) {
	return <LetterX {...restProps}>{children}</LetterX>;
};

LogoAnimation.Shape = function LogoAnimationLetter({ children, ...restProps }) {
	return (
		<Shape {...restProps}>
			<Fill />
		</Shape>
	);
};

export default LogoAnimation;
