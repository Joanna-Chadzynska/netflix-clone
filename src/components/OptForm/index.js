import React, { createContext, useContext } from 'react';
import { useForm } from '../../hooks/useForm';
import { validateForm } from '../../utils';
import {
	Break,
	Button,
	Container,
	ErrorMessage,
	Inner,
	Input,
	InputPlacement,
	Label,
	Text,
} from './styles/optForm';

const InputContext = createContext();

const OptForm = ({ children, ...restProps }) => {
	// const emailSend = () => {
	// 	// console.log('No errors, submit callback called!');
	// };

	const { values, errors, handleChange, handleSubmit } = useForm(
		console.log('No errors, submit callback called!'),
		validateForm
	);

	return (
		<Container {...restProps}>
			<InputContext.Provider value={{ values, handleChange, errors }}>
				<Inner onSubmit={handleSubmit} hasErrors={errors.email} noValidate>
					{children}
				</Inner>
			</InputContext.Provider>
		</Container>
	);
};

OptForm.Text = function OptFormText({ children, ...restProps }) {
	return <Text {...restProps}>{children}</Text>;
};

OptForm.Button = function OptFormButton({ children, ...restProps }) {
	return (
		<Button {...restProps}>
			{children}
			<img src='/images/icons/chevron-right.png' alt='Try Now' />
		</Button>
	);
};

OptForm.Input = function OptFormInput({ ...restProps }) {
	const { values, handleChange } = useContext(InputContext);

	return (
		<Input {...restProps} onChange={handleChange} value={values.email || ''} />
	);
};

OptForm.ErrorMessage = function OptFormErrorMessage({ message, ...restProps }) {
	const { errors } = useContext(InputContext);
	return (
		<ErrorMessage {...restProps} data-testid='error-message'>
			{errors.email ? errors.email : message}
		</ErrorMessage>
	);
};

OptForm.Label = function OptFormLabel({ children, ...restProps }) {
	const { errors } = useContext(InputContext);
	return (
		<Label {...restProps} hasErrors={errors.email}>
			{children}
		</Label>
	);
};
OptForm.InputPlacement = function OptFormInputPlacement({
	children,
	...restProps
}) {
	return <InputPlacement {...restProps}>{children}</InputPlacement>;
};

OptForm.Break = function OptFormTitle({ ...restProps }) {
	return <Break {...restProps} />;
};

export default OptForm;
