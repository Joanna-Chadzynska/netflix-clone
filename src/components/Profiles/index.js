import React from 'react';
import {
	Container,
	Item,
	List,
	Picture,
	Title,
	UserName,
} from './styles/profiles';

const Profiles = ({ children, ...restProps }) => {
	return <Container {...restProps}>{children}</Container>;
};

Profiles.Title = function ProfilesTitle({ children, ...restProps }) {
	return <Title {...restProps}>{children}</Title>;
};

Profiles.List = function ProfilesList({ children, ...restProps }) {
	return <List {...restProps}>{children}</List>;
};

Profiles.User = function ProfilesUser({ children, ...restProps }) {
	return <Item {...restProps}>{children}</Item>;
};

Profiles.Picture = function ProfilesPicture({ src, ...restProps }) {
	return (
		<Picture
			{...restProps}
			src={src ? `/images/users/${src}.png` : '/images/misc/loading.gif'}
			data-testid='profile-picture'
		/>
	);
};

Profiles.UserName = function ProfilesUserNAme({ children, ...restProps }) {
	return <UserName {...restProps}>{children}</UserName>;
};

export default Profiles;
