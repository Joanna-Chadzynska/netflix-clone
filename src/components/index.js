export { default as Accordion } from './Accordion';
// export * from './Card';
// export * from './Feature';
export { default as Footer } from './Footer';
// export * from './Form';
// export * from './Header';
export { default as Jumbotron } from './Jumbotron';
// export * from './Loading';
export { default as OptForm } from './OptForm';
// export * from './Player';
// export * from './Profiles';
