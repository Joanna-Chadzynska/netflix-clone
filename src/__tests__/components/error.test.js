import { render } from '@testing-library/react';
import React from 'react';
import { Error } from '../../components';
import Theme from '../../styles/theme';

jest.mock('react-router-dom');

describe('<Error/>', () => {
	it('renders component <Error/> with populated data', () => {
		const { container, queryByText } = render(
			<Theme>
				<Error>
					<Error.Inner>
						<Error.Title>Lost your way?.</Error.Title>
						<Error.Content>
							<Error.SubTitle>
								Sorry, we can't find that page. You'll find lots to explore on
								the home page.
							</Error.SubTitle>
							<Error.ButtonLink to='/Home'>Netflix Home</Error.ButtonLink>
							<Error.Code>
								<span>
									Error Code <strong>NSES-404</strong>
								</span>
							</Error.Code>
						</Error.Content>
					</Error.Inner>

					<Error.ImageSource>
						<span>
							From <strong>Lost in space</strong>
						</span>
					</Error.ImageSource>

					<Error.RequestInfo>
						<p>Build Identifier: v6591ee28</p>
						<p>Instance: i-0e688907957decfb8</p>
						<p>Request Id: d2a05002-4b20-48c4-b28c-a0529bc7f2f2-240198057</p>
					</Error.RequestInfo>
				</Error>
			</Theme>
		);

		expect(
			queryByText(
				"Sorry, we can't find that page. You'll find lots to explore on the home page."
			)
		).toBeTruthy();
		expect(queryByText('Lost your way?.')).toBeTruthy();
		expect(queryByText('Lost in space')).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
