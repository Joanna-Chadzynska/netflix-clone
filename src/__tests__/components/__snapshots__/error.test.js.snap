// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error/> renders component <Error/> with populated data 1`] = `
<div
  class="error__Background-cntjj-2 ffqOVq"
>
  <div
    class="error__Container-cntjj-0 kmIjvc"
  >
    <div
      class="error__Inner-cntjj-1 hAoZju"
    >
      <h1
        class="error__Title-cntjj-3 dyfLrx"
      >
        Lost your way?.
      </h1>
      <div
        class="error__Content-cntjj-4 hcEKBL"
      >
        <p
          class="error__SubTitle-cntjj-5 kbeEsw"
        >
          Sorry, we can't find that page. You'll find lots to explore on the home page.
        </p>
        <div
          class="error__Code-cntjj-7 iJGCSa"
        >
          <span>
            Error Code 
            <strong>
              NSES-404
            </strong>
          </span>
        </div>
      </div>
    </div>
    <span
      class="error__ImageSource-cntjj-8 kEujYS"
    >
      <span>
        From 
        <strong>
          Lost in space
        </strong>
      </span>
    </span>
    <div
      class="error__RequestInfo-cntjj-9 hIxrUj"
    >
      <p>
        Build Identifier: v6591ee28
      </p>
      <p>
        Instance: i-0e688907957decfb8
      </p>
      <p>
        Request Id: d2a05002-4b20-48c4-b28c-a0529bc7f2f2-240198057
      </p>
    </div>
  </div>
</div>
`;
